web-server/src/slices/service.ts

```diff
@@ -0,0 +1,76 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { ServiceNames } from '@/constants/service';
+
+type Status = {
+  isUp: boolean;
+};
+
+type Service = {
+  isUp: boolean;
+  logs: string[];
+};
+
+export type ServiceStatusState = {
+  [key in ServiceNames]: Service;
+};
+
+type State = {
+  services: ServiceStatusState;
+  loading: boolean;
+  error?: string;
+  active: string | null;
+};
+
+const initialState: State = {
+  services: {
+    [ServiceNames.API_SERVER]: { isUp: false, logs: [] },
+    [ServiceNames.REDIS]: { isUp: false, logs: [] },
+    [ServiceNames.POSTGRES]: { isUp: false, logs: [] },
+    [ServiceNames.SYNC_SERVER]: { isUp: false, logs: [] }
+  },
+  active: null,
+  loading: true,
+  error: undefined
+};
+
+type SetStatusPayload = {
+  statuses: { [key in ServiceNames]: Status };
+};
+
+export const serviceSlice = createSlice({
+  name: 'services',
+  initialState,
+  reducers: {
+    setActiveService: (state, action: PayloadAction<string | null>) => {
+      state.active = action.payload;
+    },
+    setLoading: (state, action: PayloadAction<boolean>) => {
+      state.loading = action.payload;
+    },
+    setStatus: (state, action: PayloadAction<SetStatusPayload>) => {
+      state.loading = false;
+      const { statuses } = action.payload;
+
+      for (const [serviceName, { isUp }] of Object.entries(statuses)) {
+        if (state.services[serviceName as ServiceNames]) {
+          state.services[serviceName as ServiceNames].isUp = isUp;
+        }
```

Comment:
can be wrapped in a much better forEach block. very unreadable. feel free to ask for help!

Reply:
rarely use `for` and `whiles`, unless they increase readability more than `forEach`, `map` and other methods provide

Reply:
Thanks I will keep this in mind

Reply:
i can reduce it a little more
```ts
 Object.entries(statuses).forEach(([serviceName, { isUp }]) => {
    const service = state.services[serviceName as ServiceNames];

    if (service) {
      service.isUp = isUp;
    }
  });
 ```

Reply:
Thanks man that's sooo good